### PR TITLE
ETH/Oasis address toggle on Txn details view

### PR DESCRIPTION
**Pull request by lubej**
## Solution

Add ETH/Oasis address toggle on transaction details view

## Links

- https://www.loom.com/share/4f7c886383024bbcacd254ae07ef7cd4
- [https://ad138b97.oasis-explorer.pages.dev/mainnet/emerald/transactions/0x6ee928...eca3dc4](https://ad138b97.oasis-explorer.pages.dev/mainnet/emerald/transactions/0x6ee9285d880724e75acef669f0d99c58f1353f8617bda9875f0750b5aeca3dc4)

### CODE REVIEW

File: src/app/components/AddressSwitch/index.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      backgroundPosition: 'center',
+      backgroundImage: `url("${oasisLogo}")`,
+    },
+  },
+  [`& .${switchClasses.track}`]: {
+    opacity: 1,
+    backgroundColor: COLORS.grayMediumDark,
+    borderRadius: '50px',
+  },
+}))
```
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/components/AddressSwitch/index.tsx:61]: How about:
```tsx
const StyledSwitch = styled(Switch)(() => ({
  zoom: 2,
  [`.${switchClasses.switchBase} .${switchClasses.thumb}`]: {
    backgroundColor: COLORS.brandMedium,
    backgroundImage: `url("${oasisLogo}")`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: '15px',
  },
  [`.${switchClasses.switchBase}.${switchClasses.checked} .${switchClasses.thumb}`]: {
    backgroundColor: COLORS.lightSilver,
    backgroundImage: `url("${ethLogo}")`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: '10px',
  },
  [`&& .${switchClasses.track}`]: {
    opacity: 1,
    backgroundColor: COLORS.grayMediumDark,
  },
}))
```

Before, After
bb

File: src/app/components/AddressSwitch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  return (
+    <StyledAddressSwitch>
+      <Typography
+        sx={{
+          color:
+            selected === AddressSwitchOption.Oasis
+              ? isMobile
+                ? COLORS.white
+                : COLORS.brandExtraDark
+              : COLORS.grayMediumDark,
```
Comment (lukaw3d) [inline src/app/components/AddressSwitch/index.tsx:96]: I don't like that implementation details of SubPageCard are leaking into AddressSwitch. Can we structure things better?
cc @csillag do you know if there's an obvious solution with themes? e.g. marking mobile SubPageCard header as dark theme?

File: src/app/components/Transactions/Logs.tsx [hunk 3]
```diff
@@ -18,20 +19,35 @@ export const TransactionLogs: FC<{
...
+      events={data?.data?.events}
+      isLoading={isLoading}
+      addressSwitchOption={addressSwitchOption}
+    />
+  )
 }
 
 export const TransactionLogsView: FC<{
-  scope: SearchScope
+  scope: SearchScope & RuntimeTransaction
```
Comment (lukaw3d) [inline src/app/components/Transactions/Logs.tsx:33]: Did you mean to change this to `transaction: RuntimeTransaction`

**Review by lukaw3d — CHANGES_REQUESTED** [review]

File: src/app/components/AddressSwitch/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/components/AddressSwitch/index.tsx]: This adds some padding to the element, which breaks the layout.

[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/components/AddressSwitch/index.tsx]: Updated, slightly more hacky:
```tsx
const StyledSwitch = styled(Switch)(() => ({
  zoom: 2,
  margin: '-8px', // Reduce padding
  [`.${switchClasses.switchBase} .${switchClasses.thumb}`]: {
    backgroundColor: COLORS.brandMedium,
    backgroundImage: `url("${oasisLogo}")`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: '15px',
  },
  [`.${switchClasses.switchBase}.${switchClasses.checked} .${switchClasses.thumb}`]: {
    backgroundColor: COLORS.lightSilver,
    backgroundImage: `url("${ethLogo}")`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: '10px',
  },
  [`&&&& .${switchClasses.track}`]: {
    opacity: 1,
    backgroundColor: COLORS.grayMedium2,
  },
}))
```

Before
after
switch

[the remark this answers is not in the document]
Comment (lubej) [reply src/app/components/AddressSwitch/index.tsx]: Considering how much it simplifies the code, did use your version 👍

File: src/app/components/Transactions/LogEvent.tsx [hunk 4]
```diff
@@ -14,8 +14,15 @@ import { AccountLink } from '../Account/AccountLink'
...
 import { SearchScope } from '../../../types/searchScope'
+import { AddressSwitchOption } from '../AddressSwitch'
+import { getOasisAddress } from '../../utils/helpers'
 
-const EvmEventParamData: FC<{ scope: SearchScope; param: EvmEventParam }> = ({ scope, param }) => {
+const EvmEventParamData: FC<{
+  scope: SearchScope
+  param: EvmEventParam
+  address?: string
+  addressSwitchOption: AddressSwitchOption
```
Comment (lukaw3d) [inline src/app/components/Transactions/LogEvent.tsx:24]: Did you mean to use addressSwitchOption somewhere in here?

File: src/app/components/Transactions/LogEvent.tsx [hunk 5]
```diff
@@ -141,7 +198,7 @@ export const TransactionLogEvent: FC<{
           <>
             <dt>{t('transaction.header')}</dt>
             <dd>
-              <TransactionLink scope={scope} hash={event.tx_hash} />
+              <TransactionLink scope={scope} hash={transactionLinkValue || transaction.hash} />
```
Comment (lukaw3d) [inline src/app/components/Transactions/LogEvent.tsx:201]: API now has event.eth_tx_hash! This can be simpler

**Review by lukaw3d — APPROVED** [review]

File: src/app/components/Transactions/LogEvent.tsx (hunk 5 above)
Comment (lubej) [reply src/app/components/Transactions/LogEvent.tsx:201]: Right good point 👍

File: src/app/components/Transactions/LogEvent.tsx (hunk 4 above)
Comment (lubej) [reply src/app/components/Transactions/LogEvent.tsx:24]: Not, nice catch.

File: src/app/pages/TransactionDetailPage/index.tsx [hunk 6]
```diff
@@ -94,6 +97,16 @@ export const TransactionDetailPage: FC = () => {
 
   const hash = useParams().hash!
 
+  const [addressSwitchOption, setAddressSwitchOption] = useState<
+    AddressSwitchOption.Oasis | AddressSwitchOption.ETH
+  >(() => {
+    if (isValidTxOasisHash(hash)) {
+      return AddressSwitchOption.Oasis
+    }
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/pages/TransactionDetailPage/index.tsx:105]: So the idea is to check if hash is Oasis type then auto switch to Oasis type. We can change this behaviour after we receive some feedback.

**Pull request merged**